scripts/Projectile.gd

```diff
@@ -1,26 +1,41 @@
 extends CharacterBody3D
 
-const BULLET_VELOCITY = 20
-const BULLET_DAMAGE = 2
 
-var time_alive = 1.5
-var hit = false
+var bullet_velocity: float
+var current_damage: float
+var projectile_lifetime: float
+var enemy_to_follow: EnemyBase
```

Comment:
Remove homing projectiles logic from base projectile, add it to a modular script to attach to instances of projectiles

Not necessary right now, opening an issue to keep track of this.